Cargo.toml

```diff
@@ -51,17 +55,22 @@ futures = "0.3.28"
 hex = { version = "0.4.3", default-features = false, features = ["alloc"] }
 hex-literal = "0.4.1"
 impl-serde = { version = "0.4.0", default-features = false }
+impl-trait-for-tuples = "0.2.1"
 jsonrpsee = "0.16.2"
 kvdb-rocksdb = "0.19.0"
 libsecp256k1 = { version = "0.7.1", default-features = false }
 log = { version = "0.4.19", default-features = false }
+num_enum = { version = "0.5.3", default-features = false }
 parity-db = "0.4.10"
 parking_lot = "0.12.1"
+paste = "1.0.6"
 rlp = { version = "0.5.2", default-features = false }
 scale-codec = { package = "parity-scale-codec", version = "3.6.4", default-features = false, features = ["derive"] }
 scale-info = { version = "2.9.0", default-features = false, features = ["derive"] }
 serde = { version = "1.0", default-features = false, features = ["derive", "alloc"] }
 serde_json = "1.0"
+sha3 = { version = "0.10", default-features = false }
```

Comment:
```suggestion
```

we could use `sp_core::hashing` directly

Reply:
done